core/src/main/java/org/elasticsearch/action/ingest/IngestActionForwarder.java

```diff
@@ -36,15 +37,18 @@
  *
  * TODO: move this into IngestService and make index/bulk actions call that
  */
-public class IngestActionForwarder {
+public final class IngestActionForwarder implements ClusterStateListener {
 
     private final ClusterService clusterService;
     private final TransportService transportService;
     private final AtomicInteger ingestNodeGenerator = new AtomicInteger(Randomness.get().nextInt());
+    private DiscoveryNode[] ingestNodes;
 
     public IngestActionForwarder(ClusterService clusterService, TransportService transportService) {
```

Comment:
can we maybe remove the dependency to ClusterService from this class it might be cleaner